Source/DafnyLanguageServer/Handlers/DafnyHoverHandler.cs

```diff
@@ -313,20 +313,34 @@ class AssertionBatchIndexComparer : IComparer<AssertionBatchIndex> {
       return information;
     }
 
-    private string FormatResourceCount(int nodeResourceCount) {
+    public static string FormatResourceCount(long nodeResourceCount) {
       var suffix = 0;
+      var fractional = 0;
+      // First, round up nodeResourceCount to 3 significant digits.
+      var nDigits = ("" + nodeResourceCount).Length;
+      if (nDigits > 3) {
+        var toRemove = (long)Math.Pow(10, nDigits - 3);
+        nodeResourceCount += toRemove / 2;
+        nodeResourceCount -= (nodeResourceCount % toRemove);
+      }
+
       while (nodeResourceCount / 1000 >= 1 && suffix < 4) {
+        fractional = (int)(nodeResourceCount % 1000);
```

Comment:
Why do you assign to fractional in the loop instead of after it?

Reply:
Because it depends on a number that is modified after this assignment.